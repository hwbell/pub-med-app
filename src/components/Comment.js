import React from 'react';
import '../App.scss';
import 'bootstrap/dist/css/bootstrap.css';

// components
import { Button } from 'reactstrap';

// ******************************************************************************
class Comment extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    let { user, text } = this.props.comment;
    
    return (
      <div className="comment">

        <p className="comment-title">{`${user} - `}</p>

        <p className="comment-text">{text}</p>

      </div>
    );
  }
}

const styles = {

}

export default Comment;
