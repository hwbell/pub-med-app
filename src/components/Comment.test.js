import React from 'react';

import Comment from './Comment';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

const someProps = {
  comment: {
    text: 'This article seems to lack source material for the conclusions drawn in the discussion.',
    user: 'Jeff'
  }
};

describe('Comment', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Comment { ...someProps}/>)
  })

  it('renders without crashing', () => {
    wrapper.update();
  });

  it('renders correctly', async () => {
    const tree = await renderer
      .create(<Comment  {...someProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the correct elements', () => {
    expect(wrapper.find('.comment').length).toBe(1);
    expect(wrapper.find('.profile-title').length).toBe(1);
    expect(wrapper.find('.paragraph').length).toBe(1);
    expect(wrapper.find('hr').length).toBe(1);
  })

})