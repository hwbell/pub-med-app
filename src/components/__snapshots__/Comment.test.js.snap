// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders correctly 1`] = `
<div
  className="comment"
>
  <p
    className="comment-title"
  >
    <i
      className="far fa-user"
    />
     Jeff says: 
  </p>
  <p
    className="comment-text"
  >
    This article seems to lack source material for the conclusions drawn in the discussion.
  </p>
</div>
`;
